irr/src/CXMeshFileLoader.cpp

```diff
@@ -965,7 +965,7 @@ bool CXMeshFileLoader::parseDataObjectSkinWeights(SXMesh &mesh)
 	u32 i;
 
 	const u32 jointStart = joint->Weights.size();
-	joint->Weights.reallocate(jointStart + nWeights);
+	joint->Weights.reserve(jointStart + nWeights);
```

Comment:
has to be `resize` since you write to `joint->Weights[jointStart + X]` below

Reply:
fixed, but i think i was preserving previous (wrong) behavior here. the irr array impl for `reallocate` is

```cpp
	//! Reallocates the array, make it bigger or smaller.
	/** \param new_size New size of array.
	\param canShrink Specifies whether the array is reallocated even if
	enough space is available. Setting this flag to false can speed up
	array usage, but may use more memory than required by the data.
	*/
	void reallocate(u32 new_size, bool canShrink = true)
	{
		size_t allocated = m_data.capacity();
		if (new_size < allocated) {
			if (canShrink)
				m_data.resize(new_size);
		} else {
			m_data.reserve(new_size);
		}
	}
```

Reply:
it's possible that this could have worked by accident, since this `reallocate` impl actually has a bug